content/blog/jlink-to-produce-own-runtime/index.md

```diff
@@ -0,0 +1,142 @@
+---
+title: Creating your own runtime using jlink
+date: "2021-10-04T12:00:00+00:00"
+author: sxa
+description: How to swap a JRE image for a jlink runtime
+tags:
+  - Temurin
+  - Docker
+---
+
+With the release of the latest LTS version of Temurin we have decided not to ship Java Runtime Environments (JREs) separately from the JDK downloads.
+This post will show you how to create your own runtime for Java 17+ which is comparable to a legacy JRE if required.
+
+## Why have you decided to stop shipping JREs?
+
+While the OpenJDK build process still has support for building a JRE via the
+`legacy-jre` target it is, as the name suggests, legacy functionality.  The
+new LTS version provided us with an opportunity to make a clean break and we
+have decided now is the time to no longer provide JREs since there are more
+efficient options available. It also simplifies the list of downloads we have
+available and reduces our testing overhead.  We will continue to produce
+them for the existing 8 and 11 versions to retain continuity for those using
+them.  The discussion on this happened in
+[this issue](https://github.com/adoptium/temurin-build/issues/2683).
+
+The good news is that it is very easy to produce your own runtime that looks
+comparable to a legacy JRE, and in many cases it can be smaller than an old
+legacy JRE would be!
+
+## Sounds good. So how do I create my own "JRE"?
+
+Creating your own runtime that is comparable to a legacy JRE is simpler
+then you might think.  Firstly, download and extract the JDK archive.
+Second, use [jlink](https://docs.oracle.com/en/java/javase/17/docs/specs/man/jlink.html)
+to create your own runtime that will be smaller, yet still provide
+equivalent functionality to the legacy JRE.  Replace `jdk-17+35` in the examples
+below with the version of Java you are working with, and replace the forward
+slashes with the path separator on your platform (e.g.  `\` for Windows):
+
+```
+   jdk-17+35/bin/jlink --add-modules ALL-MODULE-PATH --output jdk-17+35-jre \
+      --strip-debug --no-man-pages --no-header-files --compress=2
+```
+
+Once you have done that, the contents of the `jdk-17+35-jre` will work in
+place of a legacy JRE and be significantly smaller expanded on disk than
+either the JDK or legacy JRE would be.
+
+## What are the disk space savings?
+
+It will vary by platform and the version of Java you are using, but using
+one example with the `jdk-17+35` release, the full JDK for one platform is
+about 312Mb on disk.  The jlinked runtime using the above command is about
+95Mb.
+
+## Is this identical to the legacy JRE?
+
+Not quite. By default there are some extra modules included via the
+`ALL-MODULE-PATH` list than would be included in a normal JRE. You can list
```

Comment:
"... would be included in a legacy JRE that you might be used to." What's a "normal" JRE? :)

Reply:
Changd to "in the legacy JRE" for consistency with the rest of the post now.